src/foam/nanos/auth/CapabilityAuthService.js

```diff
@@ -0,0 +1,134 @@
+foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'CapabilityAuthService',
+  extends: 'foam.nanos.auth.ProxyAuthService',
+  documentation: `
+  this decorator checks for either a capability or permission string. If the check returns false, delegate to next authservice. Return true otherwise.
+  `,
+
+  implements: [
+    'foam.nanos.auth.AuthService',
+  ],
+
+  javaImports: [
+    'foam.dao.ArraySink',
+    'foam.dao.DAO',
+    'foam.nanos.crunch.Capability',
+    'foam.nanos.crunch.CapabilityJunctionStatus',
+    'foam.nanos.crunch.UserCapabilityJunction',
+    'foam.nanos.logger.Logger',
+    'foam.nanos.session.Session',
+    'java.util.List',
+    'javax.security.auth.AuthPermission',
+    'static foam.mlang.MLang.*'
+  ],
+
+  imports: [
+
+  ],
+
+  methods: [
+    {
+      name: 'check',
+      documentation: `
+      check if the given input string is in the userCapabilityJunctions or implied by a capability in userCapabilityJunctions for the current context user
+      `,
+      javaCode: `
+      if ( x == null || permission == null ) return false;
+      if ( x.get(Session.class) == null ) return false;
+      User user = (User) x.get("user");
+      if( user == null || ! user.getEnabled() ) return false;
+
+      // temporary fix to get around authservice being called on build before some services (userCapabilityJunctionDAO) are available
+      if ( user.getId() == 1 ) return true;  
+
+      if ( ! getDelegate().check(x, "service.auth.checkUser") ) throw new AuthorizationException();
+
+      try {
+        DAO userCapabilityJunctionDAO = (DAO) x.get("userCapabilityJunctionDAO");
+
+        if(userCapabilityJunctionDAO.find(
+          AND(
+            EQ(UserCapabilityJunction.SOURCE_ID, user.getId()),
+            EQ(UserCapabilityJunction.TARGET_ID, permission),
+            EQ(UserCapabilityJunction.STATUS, CapabilityJunctionStatus.GRANTED)
+          )) != null) return true;
+        
+        List<UserCapabilityJunction> userCapabilityJunctions = (List<UserCapabilityJunction>) ((ArraySink) userCapabilityJunctionDAO
```

Comment:
Doesn't the relationship create a helper method for finding all of the Capabilities for you?